app/Repositories/ProfileRepository.php

```diff
@@ -9,44 +10,50 @@
 class ProfileRepository
 {
     protected $model;
+
     public function __construct(Profile $model)
     {
         $this->model = $model;
     }
 
-    public function getProfile($request, $address) {
-        return $this->model->with(['users.circle.protocol','users.teammates','users.histories.epoch'])->byAddress($address)->first();
+    public function getProfile($address, $relationships = [])
+    {
+        return $this->model->with($relationships)->byAddress($address)->first();
     }
 
-    public function saveProfile($request) {
-        $data = $request->only('skills','bio','telegram_username',
-            'discord_username','twitter_username','github_username','medium_username','website');
+    public function saveProfile($request)
+    {
+        $data = $request->only('skills', 'bio', 'telegram_username',
+            'discord_username', 'twitter_username', 'github_username', 'medium_username', 'website');
         $profile = $request->profile;
         $profile->update($data);
-        $profile->load(['users.circle.protocol','users.teammates','users.histories.epoch']);
+        $profile->load(['users.circle.protocol', 'users.teammates', 'users.histories.epoch']);
         return $profile;
     }
 
-    public function uploadProfileAvatar($request, $address) {
-        $profile =  $this->model->byAddress($address)->first();
-        if($profile) {
+    public function uploadProfileAvatar($request, $address = null)
+    {
+        $profile = $address ? $this->model->byAddress($address)->first() : $request->user();
```

Comment:
we don't allow uploading profile images for others, right? so it's ok to completely drop the $address arg

Reply:
removed this